docs/howto/upgrade_from_previous_versions.md

```diff
@@ -0,0 +1,105 @@
+# How to upgrade from previous versions
```

Comment:
I think `UPGRADE.md` or `UPGRADING.md` yes caps might be a better name.

Also, would be good to add to the gemspec

`spec.upgrade = 'See docs/howto/UPGRADE.md for upgrade instructions` (syntax might not be quite right)